packages/host/app/components/matrix/room-message.gts

```diff
@@ -333,7 +352,40 @@ export default class RoomMessage extends Component<Signature> {
     }
   });
 
-  private get previewPatchCode() {
+  private patchCard = task(async (eventId: string, payload: PatchObject) => {
+    if (this.operatorModeStateService.patchCard.isRunning) {
+      return;
+    }
+    //TODO: The discriminant doesn't interact with at the field level
+    this.matrixService.failedCommandState.delete(eventId);
+    return await this.operatorModeStateService.patchCard.perform(
+      payload.id,
+      payload.patch,
+    );
+  });
+
+  private searchCard = task(async (payload: SearchObject) => {
+    let query = payload.search;
+    let adoptsFrom = {
+      name: query.filter.type.name,
+      module: query.filter.type.module,
+    };
+    //If ai returns a relative module path, we make it absolute
+    let maybeCodeRef = codeRefWithAbsoluteURL(
```

Comment:
The AI is just not good at understanding URLS, so it usually returns a relative URL which we resolve based upon the card that is in view. Suggestions welcomed